### PR TITLE
fix(signin): Hide the Unauthorized avatar error

### LINKED ISSUE #2481: Avatars: displayAccountProfileImage() throws an Unauthorized error on /signin
image

### PR DESCRIPTION
**Pull request by eoger**
Fixes #2481

### CODE REVIEW

**Comment by TDA** [comment]
LGTM! lets wait for @vladikoff to merge it!

**Comment by vladikoff** [comment]
@zaach assigned you because you might have an opinion how this error should be handled

File: app/scripts/views/mixins/avatar-mixin.js [hunk 1]
```diff
@@ -33,7 +34,9 @@ define([
           self._updateCachedProfileImage(account.get('uid'), profileImage);
           return profileImage;
         }, function (err) {
-          self.logError(err);
+          if (err.errno !== ProfileErrors.ERRORS.UNAUTHORIZED.errno) {
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/avatar-mixin.js]: You should be able to do `if (ProfileErrors.is(err, 'UNAUTHORIZED')) {` and that'll be a bit cleaner.

File: app/scripts/views/mixins/avatar-mixin.js [hunk 2]
```diff
@@ -5,8 +5,9 @@
 // helper functions for views with a profile image. Meant to be mixed into views.
 
 define([
-  'models/profile-image'
-], function (ProfileImage) {
+  'models/profile-image',
+  'lib/profile-errors'
```
Comment (shane-tomlinson) [inline app/scripts/views/mixins/avatar-mixin.js]: Can you swap the order of these? I'm going to open a PR one of these days to alphabetize the dependencies, unless your webpack PR lands first.

**Comment by eoger** [comment]
Thank you Shane!

**Comment by zaach** [comment]
IIUC, these errors occur when we attempt to fetch a profile image for an ephemeral, non-authenticated `account` instance, e.g. on /signin when there are no cached accounts. Another approach we can take so that we don't ignore legit errors is to check if the account has a `sessionToken` before attempting to fetch a profile image. If it doesn't, we can simply return a default profile image (perhaps modifying [this](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/models/account.js#L180) method).

**Comment by zaach** [comment]
Actually, after further investigation this might be the best course of action for now. We currently don't clear the accessToken if it goes bad, so these errors could still be spammed. The accessToken issue should be addressed in #2785.

**Comment by zaach** [comment]
:eyes: This PR is missing a unit test.

**Comment by vladikoff** [comment]
ping @eoger

**Comment by eoger** [comment]
Tests added

**Comment by zaach** [comment]
Nice!

**Pull request merged**